overlord/ifacestate/handlers.go

```diff
@@ -336,6 +333,40 @@ func (m *InterfaceManager) undoDiscardConns(task *state.Task, _ *tomb.Tomb) erro
 	return nil
 }
 
+func getTaskHookAttributes(task *state.Task) (map[string]interface{}, map[string]interface{}, error) {
+	var plugAttrs, slotAttrs map[string]map[string]interface{}
+
+	var err error
+	if err = task.Get("plug-attrs", &plugAttrs); err != nil {
+		return nil, nil, err
+	}
+	err = task.Get("slot-attrs", &slotAttrs)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return plugAttrs["dynamic"], slotAttrs["dynamic"], nil
+}
+
+func setTaskHookAttributes(task *state.Task, dynamicPlugAttrs map[string]interface{}, dynamicSlotAttrs map[string]interface{}) error {
+	var plugAttrs, slotAttrs map[string]interface{}
+
+	if err := task.Get("plug-attrs", &plugAttrs); err != nil {
+		return err
+	}
+	if err := task.Get("slot-attrs", &slotAttrs); err != nil {
+		return err
+	}
+
+	plugAttrs["dynamic"] = dynamicPlugAttrs
+	slotAttrs["dynamic"] = dynamicSlotAttrs
+
+	task.Set("plug-attrs", plugAttrs)
+	task.Set("slot-attrs", slotAttrs)
```

Comment:
Similar note for these functions. "plug-static" and "plug-dynamic" would simplify them.